src/internal/components/button-trigger/index.tsx

```diff
@@ -103,8 +108,16 @@ const ButtonTrigger = (
     attributes['aria-invalid'] = invalid;
   }
 
+  const analyticsMetadata: GeneratedAnalyticsMetadataButtonTriggerExpand = {
+    action: 'expand',
+    detail: {
+      label: { root: 'self' },
+      expanded: `${!pressed}`,
+    },
+  };
+
   return (
-    <button ref={ref} {...attributes}>
+    <button ref={ref} {...attributes} {...(disabled ? {} : getAnalyticsMetadataAttribute(analyticsMetadata))}>
```

Comment:
Shouldn't it include readonly? When Select is readonly the the dropdown cannot be opened.

Reply:
Nicely spotted, addressed!